src/components/Header/index.tsx

```diff
@@ -82,6 +82,9 @@ const Header = ({
               <button
                 type="button"
                 className="dark-mode-toggle"
+                onKeyPress={e =>
```

Comment:
Please fix the linting warning.
Please extract this inline function into a named function.
This only works with enter. The original issue was for enter or space keypresses. 
Please add a test for this new functionality

Reply:
Fixed all except the test part. I can't figure out a way to test a functionality like this and need some help there.